### PR TITLE
fix(components): fix ml_engine/train interface

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
[ai_platform.ipynb](https://github.com/kubeflow/pipelines/blob/master/samples/core/ai_platform/ai_platform.ipynb) is broken with the following runtime error:
```
Fire trace:
1. Initial component
2. Accessed property "train" (kfp_component/google/ml_engine/_train.py:18)
3. Called routine "train" (kfp_component/google/ml_engine/_train.py:18)
4. ('Cannot find target in dict:', '--job_dir_output_path', {u'trainingOutput': {u'consumedMLUnits': 0.06}, u'trainingInput': {u'runtimeVersion': u'1.13', u'region': u'us-central1', u'args': [u'--data-file-url', u'gs://chesu-kfp/reports.csv', u'--job-dir', u'gs://chesu-kfp'], u'pythonModule': u'trainer.task', u'jobDir': u'gs://chesu-kfp/train/output/1601963201/', u'packageUris': [u'gs://chicago-crime/chicago_crime_trainer-0.0.tar.gz']}, u'jobId': u'job_d59abdaca2e22615a2d4916a3072ac16', u'state': u'SUCCEEDED', u'etag': u'2sK+UnbUkAI=', u'startTime': u'2020-10-06T05:47:45Z', u'endTime': u'2020-10-06T05:53:48Z', u'createTime': u'2020-10-06T05:47:10Z'})
```
In #4123, `--job_dir_output_path, {outputPath: job_dir},` was added to components/gcp/ml_engine/train/component.yaml, but the `job_dir_output_path` is missing from the interface of ml_engine/train.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [x] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

/assign @Ark-kun @numerology

### CODE REVIEW

File: components/gcp/container/component_sdk/python/kfp_component/google/ml_engine/_train.py [hunk 1]
```diff
@@ -15,45 +15,60 @@
...
+
 @decorators.SetParseFns(python_version=str, runtime_version=str)
-def train(project_id, job_id_output_path, python_module=None, package_uris=None, 
-    region=None, args=None, job_dir=None, python_version=None, 
-    runtime_version=None, master_image_uri=None, worker_image_uri=None, 
-    training_input=None, job_id_prefix=None, job_id=None, wait_interval=30):
+def train(project_id,
+          job_id_output_path,
+          job_dir_output_path,
+          python_module=None,
```
Comment (numerology) [inline components/gcp/container/component_sdk/python/kfp_component/google/ml_engine/_train.py:23]: Not urgent but weird... I vaguely remember the google style should be `python_module = None`. What's the linter are you using for this file?

**Review by numerology — COMMENTED** [review]
Thanks!

File: components/gcp/container/component_sdk/python/kfp_component/google/ml_engine/_train.py (hunk 1 above)
Comment (chensun) [reply components/gcp/container/component_sdk/python/kfp_component/google/ml_engine/_train.py:23]: I used go/pyformat, and it didn't add the space. Guessing either the space is not a hard requirement or it's only enforced when there's type hints?

Comment (numerology) [reply components/gcp/container/component_sdk/python/kfp_component/google/ml_engine/_train.py:23]: SG.

**Comment by Ark-kun** [comment]
Thank you for debugging and fixing this.
/lgtm
/approve

**Comment by chensun** [comment]
kubeflow-pipeline-integration-test is constantly failing with the same error, happened during testing [ai_platform.ipynb](https://github.com/kubeflow/pipelines/blob/master/samples/core/ai_platform/ai_platform.ipynb)

I just realize that this test is only triggered on demand. I'm trying to reproduce the failure with a dummy change #4596, and I think it will.
If the failure can be reproduced with that dummy PR. I plan to force merge this PR. Because I don't think this change will affect that notebook yet -- The change hasn't been built into a container, while that notebook is running with an old image. So my guess is the integration test has been failed already. And the fix is probably to update the container image once this change merged. WDYT? @numerology @Ark-kun

**Comment by numerology** [comment]
> The change hasn't been built into a container

Yes you're right. Sorry I forgot the change needs to be built into container first.

**Comment by Ark-kun** [comment]
This PR should fix the AI Platform issue: https://github.com/kubeflow/pipelines/pull/4597

**Comment by chensun** [comment]
> This PR should fix the AI Platform issue: #4597

Thanks, Alexey. I spotted that error but thought It was one of those ignorable error. The notebook still won't work without patching this change into a new docker image as I manually tested. But maybe our tests doesn't check that far? Let me rerun the test and see.

/test kubeflow-pipeline-integration-test

**Pull request merged**